### ISSUE TITLE
Automatically apply changes to project when`twa-manifest.json` was changed.

### ISSUE THREAD

**Issue by andreban** [GoogleChromeLabs/bubblewrap#368]
**Use-Case:**

1. Developer initialises the application, the project is generated and the configuration stored at `twa-manifest.json`. 
2. Developer changes a configuration in `twa-manifest.json`.
3. Developers runs `bubblewrap build`.

**The problem:**
The developer wasn't aware of/forgot running `bubblewrap update`. So, the application doesn't reflect the latest changes to `twa-manifest.json`.

**Proposal:**
When the project is generated, store the checksum for `twa-manifest.json`. When build is invoked, read the current TWA Manifest and if the checksum doesn't match, ask the developer if they want to apply the changes to the project before building.

Alternatively, use the timestamp from the files to compare.

**Comment by joycetoh8** [comment]
I want to start implementing this feature. I was thinking of storing the checksum in `twa-manifest.json` itself (first or last line of the file) and when generating the checksum, ignore the line where the checksum is stored. Will this work or is there another option?

**Comment by andreban** [comment]
I think the checksums can be stored in a separate file?

**Comment by PEConn** [comment]
Potentially we could store the TWA manifest of the previous (or all previous) versions, so, eg, we'd end up with:

```
./twa-manifest.json
...
./previousManifests/twa-manifest-v1.json
./previousManifests/twa-manifest-v2.json
./previousManifests/twa-manifest-v3.json
```

(or just a single `previous-twa-manifest.json` or something)

This way we could do a bit more of an intelligent comparison between the last manifest and the current one - eg, so we ignore formatting changes.

(If we knew that the developers would be storing their twa-manifest in a git repo, we could probably get even more fancy - but that's probably over complicating things.)

**Comment by joycetoh8** [comment]
Thank you both for sharing your thoughts. I have a branch [here](https://github.com/GoogleChromeLabs/bubblewrap/compare/auto-apply-twa-manifest-changes) where I've tried to implement the checksum stored in a separate file method. Please take a look and I'd like to know what you think!

I like Peter's idea about storing the actual manifest so we can do a better comparison, so I may look into that as well.

**Comment by andreban** [comment]
The approach in the branch looks good to me. One note is that instead of importing and executing the `update` command directly, I'd prefer to have a shared method that is used by both `build` and `update`. I think update already has a bit of that.

**Comment by joycetoh8** [comment]
@andreban Yes I would agree! I didn't like calling the `update` command with the empty args, so I'll implement what you explained there.

@PEConn sounds good, thank you! :)

**Comment by joycetoh8** [comment]
PR #528 is up for your review :)

**Comment by joycetoh8** [comment]
Closed by #528
